### PR TITLE
UCT/API/CUDA/INFO: Implement dmabuf export

### PR DESCRIPTION
**Pull request by yosefe**
## Why
- Support dmabuf export - querying dmabuf_fd and dmabuf_offset

## How
- Extend uct_mem_query() API
- Implemend dmabuf query for cuda_copy transport
- Add dmabuf capabilities print to ucx_info

Replaces #7847

### CODE REVIEW

**Review by hoopoepg — DISMISSED** [review]

**Comment by yosefe** [comment]
@Akshay-Venkatesh  can you pls review?

**Review by brminich — APPROVED** [review]

**Review by Akshay-Venkatesh — APPROVED** [review]

File: src/uct/cuda/cuda_copy/cuda_copy_md.c [hunk 1]
```diff
@@ -44,6 +47,34 @@ static ucs_config_field_t uct_cuda_copy_md_config_table[] = {
     {NULL}
 };
 
+static int uct_cuda_copy_md_is_dmabuf_supported()
+{
+    int dmabuf_supported = 0;
+    CUdevice cuda_device;
+    CUresult cu_err;
+
+    cu_err = cuDeviceGet(&cuda_device, 0);
```
[the remark this answers is not in the document]
Comment (yosefe) [reply src/uct/cuda/cuda_copy/cuda_copy_md.c:56]: @Akshay-Venkatesh fixed

**Comment by yosefe** [comment]
@tonycurtis @shamisp  can you pls review the documentation?

File: docs/source/faq.md [hunk 2]
```diff
@@ -358,11 +358,19 @@ setting `UCX_MEMTYPE_CACHE=n`.
 #### Does UCX support zero-copy for GPU memory over RDMA?
 
 Yes. For large messages UCX can transfer GPU memory using zero-copy RDMA using
-rendezvous protocol. It requires the peer memory q for the relevant GPU type
-to be loaded on the system.
+rendezvous protocol. It requires a peer memory driver for the relevant GPU type
+o be loaded, or (starting with UCX v1.14) dmabuf support on the system.
```
Comment (tonycurtis) [inline docs/source/faq.md:362]: to be loaded

File: src/uct/api/uct.h [hunk 3]
```diff
@@ -1595,6 +1610,22 @@ typedef struct uct_md_mem_attr {
      * to uct_md_mem_query is returned as is.
      */
     size_t            alloc_length;
+
+    /**
+     * Dmabuf file descriptor to expose memory regions across devices. Refer
+     * (https://01.org/linuxgraphics/gfx-docs/drm/driver-api/dma-buf.html).
+     * If the md does not support querying fd object associated with the region,
```
Comment (tonycurtis) [inline src/uct/api/uct.h:1617]: the fd object

File: src/uct/api/uct.h [hunk 4]
```diff
@@ -1595,6 +1610,22 @@ typedef struct uct_md_mem_attr {
...
+     * then dmabuf_fd is set to UCT_DMABUF_FD_INVALID by uct_md_mem_query(). It
+     * is the responsibility of the user to close returned fd once used using
```
Comment (tonycurtis) [inline src/uct/api/uct.h:1619]: the returned fd

File: src/uct/api/uct.h [hunk 5]
```diff
@@ -1595,6 +1610,22 @@ typedef struct uct_md_mem_attr {
...
+     * close (2).
```
Comment (tonycurtis) [inline src/uct/api/uct.h:1620]: "used using" needs different phrasing

**Review by tonycurtis — CHANGES_REQUESTED** [review]

File: docs/source/faq.md [hunk 6]
```diff
@@ -358,11 +358,19 @@ setting `UCX_MEMTYPE_CACHE=n`.
...
-rendezvous protocol. It requires the peer memory q for the relevant GPU type
-to be loaded on the system.
+rendezvous protocol. It requires a peer memory driver for the relevant GPU type
+to be loaded, or (starting with UCX v1.14) dmabuf support on the system.
+
 > **NOTE:** In some cases if the RDMA network device and the GPU are not on
 the same NUMA node, such zero-copy transfer is inefficient.
 
+#### What is needed for dmabuf support?
+- UCX v1.14 or later.
```
[the remark this answers is not in the document]
Comment (yosefe) [reply docs/source/faq.md:368]: why? it's a version number, not a branch name

[the remark this answers is not in the document]
Comment (shamisp) [reply docs/source/faq.md:368]: If you insist - > 1.14.0

[the remark this answers is not in the document]
Comment (yosefe) [reply docs/source/faq.md]: done

**Review by shamisp — DISMISSED** [review]

**Comment by yosefe** [comment]
@Akshay-Venkatesh can you pls take a look?

**Review by Akshay-Venkatesh — DISMISSED** [review]

**Pull request merged**